### PR TITLE
datasource:String -> sources: String[]

### PR DESCRIPTION
**Pull request by 123joshuawu**
Refactor graphql models
Refactor all `datasource` related GUI to `sources`

### CODE REVIEW

File: gui/src/shared/components/kg/node/KgNodeLink.tsx [hunk 1]
```diff
@@ -5,13 +5,13 @@ import {kgId} from "shared/api/kgId";
...
-  datasource?: string;
-}> = ({node, datasource}) => {
+  sources?: string[];
+}> = ({node, sources}) => {
   return (
     <Link data-cy="node-link" to={Hrefs.kg({id: kgId}).node({id: node.id})}>
       {(node.label ? node.label : node.id) +
         (node.pos ? " (" + node.pos + ")" : "") +
-        (datasource ? " - " + datasource : "")}
+        (sources && sources.length > 0 ? " - " + sources.join(",") : "")}
```
Comment (gordom6) [inline gui/src/shared/components/kg/node/KgNodeLink.tsx:14]: Nit: should probably use | as the join string to be consistent with KGTK conventions.

File: gui/src/shared/components/kg/search/KgSourceLink.tsx [hunk 2]
```diff
@@ -4,18 +4,20 @@ import * as React from "react";
...
     to={Hrefs.kg({id: kgId}).nodeSearch({
-      filters: {sources: {include: [datasource]}},
+      filters: {
+        sources: {include: typeof source === "string" ? [source] : source},
+      },
       __typename: "KgNodeSearchVariables",
     })}
   >
-    {datasource}
+    {source}
```
Comment (gordom6) [inline gui/src/shared/components/kg/search/KgSourceLink.tsx:21]: Won't work if source is a string[]? Probably source should just be a string, one link per source. The "combined" data sources (post merge) will be `string`.

Comment (123joshuawu) [reply gui/src/shared/components/kg/search/KgSourceLink.tsx:21]: Hm thats true, I missed that. I will change source to be a string.

File: gui/src/shared/components/kg/node/KgNodeLink.tsx (hunk 1 above)
Comment (123joshuawu) [reply gui/src/shared/components/kg/node/KgNodeLink.tsx:14]: Changed to |

**Pull request merged**